test/utils.sh

```diff
@@ -1080,3 +1080,80 @@ get_highest_version_from_bundles_list() {
 
   echo "$highest_image"
 }
+
+# This function will be used by tasks in tekton-integration-catalog
+# Given the output of 'opm render $bundle_image' command, this function returns the suggested namespace for the bundle
+get_bundle_suggested_namespace() {
+  local RENDER_OUT_BUNDLE="$1"
+
+  # Validate input parameters
+  if [[ -z "$RENDER_OUT_BUNDLE" ]]; then
+    echo "get_bundle_suggested_namespace: Invalid input. Usage: get_bundle_suggested_namespace <RENDER_OUT_BUNDLE>" >&2
+    exit 2
+  fi
+
+  local namespace
+  namespace=$(echo "$RENDER_OUT_BUNDLE" | tr -d '\000-\031' | jq -r '
+    .properties[]? 
+    | select(.type == "olm.csv.metadata") 
+    | .value.annotations["operatorframework.io/suggested-namespace"]')
+
+  # Ensure namespace is not empty
+  if [[ -z "$namespace" || "$namespace" == "null" ]]; then
+    echo "Error: No suggested namespace found in bundle" >&2
+    exit 1
+  fi
+
+  echo "$namespace"
+}
+
+# This function will be used by tasks in tekton-integration-catalog
+# Given the output of 'opm render $bundle_image' command, this function returns the supported install modes for the bundle
+get_bundle_install_modes() {
+  local RENDER_OUT_BUNDLE="$1"
+
+  # Validate input parameters
+  if [[ -z "$RENDER_OUT_BUNDLE" ]]; then
+    echo "get_bundle_install_modes: Invalid input. Usage: get_bundle_install_modes <RENDER_OUT_BUNDLE>" >&2
+    exit 2
+  fi
+
+  local install_modes
+  install_modes=$(echo "$RENDER_OUT_BUNDLE" | tr -d '\000-\031' | jq -r '
+    .properties[]? 
+    | select(.type == "olm.csv.metadata") 
+    | .value.installModes[]? 
+    | select(.supported == true) 
+    | .type')
+
+  # Ensure install modes are not empty
+  if [[ -z "$install_modes" ]]; then
+    echo "Error: No supported install modes found in bundle" >&2
```

Comment:
Same thought as my previous comment. Maybe provide more info if there the fragment hasn't yet implemented  `olm.csv.metadata` .

Reply:
Fixed that, please take a look